webpack/assets/javascripts/react_app/components/Editor/EditorHelpers.js

```diff
@@ -0,0 +1,156 @@
+} from 'lodash';
+
+import { setCompleters } from 'ace-builds/src-noconflict/ext-language_tools';
+
+const resolveDefault = ({ default: defValue }) => {
+  switch (typeof defValue) {
+    case 'object':
+      if (defValue === null) return 'nil';
+
+      return '{}';
+    case 'string':
+      if (defValue === '') return '""';
+
+      return `"${defValue}"`;
+    default:
+      return defValue;
+  }
+};
+
+const methodReturns = ({ returns: { object: retObject } }) => {
+  // TODO: add support for Hash return object with nested description
+  // can be skipped for now since we don't use it yet and it wasn't tested properly
+  if (retObject.class === 'Hash' && retObject.data.constructor === Array)
+    return 'Object';
+  const retMeta = capitalize(replace(retObject.meta, '_', ' '));
+  const retData = replace(JSON.stringify(retObject.data), 'null', 'nil');
+  const returns = retObject.data === null ? retObject.class : retData;
+  return `${retMeta}: ${returns}`;
+};
+
+const methodTooltip = (method, isMacro) => {
+  const usagePrefix = isMacro ? '' : 'obj.';
+  const usage = `Usage:\n\t${usagePrefix}${methodSignature(method)}`;
+  const returns = `Returns:\n\t${methodReturns(method)}`;
+  if (isEmpty(method.examples))
+    return [method.short_description, usage, returns].join('\n\n');
+
+  let examples = method.examples
+    .map(e => {
+      const desc = e.desc ? `${e.desc}\n\n` : '';
+      return `${desc}${e.example}`;
+    })
+    .join('\n\n');
+  examples = `Examples:\n\n${examples}`;
+  return [method.short_description, usage, returns, examples].join('\n\n');
+};
+
+const methodSignature = (method, isSnippet) => {
+  if (isEmpty(method.params)) return method.name;
+
+  let currParam = 1;
+  const params = method.params.map(p => {
+    switch (p.type) {
+      case 'required':
+        return isSnippet ? `\${${currParam++}:${p.name}}` : p.name;
+      case 'optional': {
+        if (p.expected_type === 'list') {
+          return isSnippet
+            ? `\${${currParam++}:first},\${${currParam++}:second}`
+            : `*${p.name}`;
+        }
+        const defaultVal = resolveDefault(p);
+        return isSnippet
+          ? `\${${currParam++}:${defaultVal}}`
+          : `${p.name} = ${defaultVal}`;
+      }
+      case 'keyword':
+        return isSnippet
+          ? `${p.name}: \${${currParam++}:${resolveDefault(p)}}`
+          : `${p.name}: ${resolveDefault(p)}`;
+      default:
+        return null;
+    }
+  });
+  const filtered = filter(params, p => p !== null).join(', ');
+  const blockParam = find(method.params, p => p.type === 'block');
+  const parts = [method.name];
+  if (!isEmpty(method.params)) parts.push(`(${filtered})`);
+  if (blockParam) parts.push(` ${blockParam.schema}`);
+  return parts.join('');
+};
+
+export const parseDocs = cache => {
+  // Do nothing if the editor is not being used for templates
+  if (!cache) return;
+  // Custom basic snippets
+  /* eslint-disable */
+  let completions = [
```

Comment:
Just an idea: How about `json` file with these completions and parse it here from the file. 
If there would be a `json` file it would be a little bit more easily extendable & editable than `EditorHelpers.js` file

Reply:
Well, I thought about that, but I didn't find json as a user-friendly way to add things even if it's well formatted, yaml would be a better way to go. Moreover, I don't think those basic snippets will be changed/updated often. But if there will be more interest in this, I'll do that :)